immutable_tree.go

```diff
@@ -149,8 +149,7 @@ func (t *ImmutableTree) Has(key []byte) (bool, error) {
 
 // Hash returns the root hash.
 func (t *ImmutableTree) Hash() ([]byte, error) {
-	hash, _, err := t.root.hashWithCount()
-	return hash, err
+	return t.root.hashWithCount(t.version + 1)
```

Comment:
why + 1?

Reply:
This version is for non-persisted nodes, the current version is not updated yet.